dlt/pipeline/pipeline.py

```diff
@@ -856,29 +856,25 @@ def _extract_source(self, storage: ExtractorStorage, source: DltSource, max_para
         source_schema = source.schema
         source_schema.update_normalizers()
 
-        extract_id = extract_with_schema(storage, source, source_schema, self.collector, max_parallel_items, workers)
-
         # if source schema does not exist in the pipeline
         if source_schema.name not in self._schema_storage:
-            # save schema into the pipeline
-            self._schema_storage.save_schema(source_schema)
+            # create a new schema with the right name
+            self._schema_storage.save_schema(Schema(source_schema.name))
```

Comment:
we are we saving empty schema? I get that we update it later but I'd just save source schema and assign pipeline schema to source schema. and do the lines below in else

Reply:
I thought it would be nice to always start with an empty schema and then merge in incoming stuff, this will be easier for implementing the contracts too. There already is table stuff on the source schema before it runs through extract. But for now I have set it back to the way it was.

Reply:
you still save an empty schema but let's try that :)

Reply:
Yes, I prefer that, because the update path will always be the same, it seems cleaner to me. HOWEVER: this one test in common "tests/extract/test_incremental.py::test_replace_resets_state"n fails if I create a new schema her, so I am doing it the old way again now. I had the same problem / question on the contracts PR where this happened and I did not track it down what was going on. It seems that there is a difference between saving the schema directly on the first load and merging it into an empty one.